packages/framer-motion/src/value/use-spring.ts

```diff
@@ -35,6 +35,12 @@ export function useSpring(
     const activeSpringAnimation = useRef<PlaybackControls | null>(null)
     const value = useMotionValue(isMotionValue(source) ? source.get() : source)
 
+    const stopAnimation = () => {
+        if (activeSpringAnimation.current) {
+            activeSpringAnimation.current.stop()
+        }
+    }
```

Comment:
This is always going to be a new function, so this will negate the benefit of `useMemo` completely, right?

Reply:
No. `useMemo` (and `useCallback`, `useEffect`) always also create new functions.

In this particular instance it's being used in strictly off the book way, attaching a handler on the initial render. Given this is impure it would probably be better to change this to `useInsertionEffect`

Reply:
But what I meant is that you add a value that is always different to the dependency list of `useMemo` which  will never memoize the function result.

Reply:
`stopAnimation`? That's being sent to `attach` not the `useMemo`. Btw I also changed `useMemo` to `useInsertionEffect` as it's a more normal place to add a listener.